File: typescript/src/renderer.ts
import { areaR, getterSet } from "./shapes/rectangle.js";
import { areaS } from "./shapes/square.js";
import { areaC } from "./shapes/circle.js";


function draw(x: number) {

    const Oldnum: number = getterSet.getHeight()
    console.log("current height: " ,Oldnum);
    console.log("Shape rect drawn\n" + "Its area is " + areaR.area())
    getterSet.setHeight(500);
    const Newnum: number = getterSet.getHeight()
    console.log("new height: " ,Newnum);

    switch (x) {
        case 1:
            console.log("Shape rect (new) drawn\n" + "Its area is " + areaR.area())
            break;
        case 2:
            console.log("Shape square drawn\n" + "Its area is " + areaS.area())
            break;
        case 3:
            console.log("Shape Circle drawn\n" + "Its area is " + areaC.area())
            break;
        default:
            break;
    }
}

export { draw }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer{
    Rectangle rectangle = new Rectangle();
    Square square = new Square();
    Circle circle = new Circle();
    
    void draw(int num) {

        // assume implementation
        rectangle.setHeight(100);

        switch (num) {
            case 1:
                double areaR = rectangle.area();
                System.out.println("Shape rect printed\n" + "Its area is " + areaR);
                break;
            case 2:
                double areaS = square.area();
                System.out.println("Shape square printed\n" + "Its area is " + areaS);
                break;
            case 3:
                double areaC = circle.area();
                System.out.println("Shape circle printed\n" + "Its area is " + areaC);
                break;
            default:
                System.out.println("invalid case");
                break;
        }
    }
}
